src/Rover.cpp

```diff
@@ -153,16 +192,34 @@ int rover_loop(int argc, char **argv)
     Globals::opts = ParseCommandLineOptions(argc, argv);
     InitializeRover(MOTOR_UNIT_MODE_PWM, true);
     CANPacket packet;
-    // Target location for autonomous navigation
-    // Eventually this will be set by communication from the base station
-    int urc_leg = 5;
-    Autonomous autonomous(getLeg(urc_leg), CONTROL_HZ);
+	// Target locations for autonomous navigation
+	// Eventually this will be set by communication from the base station
+	std::queue<URCLeg> urc_legs = parseGPSLegs("../src/gps_legs.txt");
+    Autonomous autonomous(urc_legs, CONTROL_HZ);
     char buffer[MAXLINE];
     struct timeval tp_rover_start;
     int num_can_packets = 0;
     gettimeofday(&tp_rover_start, NULL);
     for(int iter = 0; /*no termination condition*/; iter++)
     {
+		// Wait for confirmation to start the next leg
+        if (Globals::AUTONOMOUS && autonomous.currentLegDone() && autonomous.hasNextLeg()) {
+			// For simulator only, change or comment this out when running on real rover
+			// Wait for user to press enter, then start the next leg
+			// TODO the real rover might want to get this command from the base station
+			std::cout << "Press enter to start next leg" << std::endl;
+			// If it takes longer than 10ms to find a newline, it was probably just inputted and we can proceed
+			long elapsedTime = 0;
+			while (elapsedTime < 10 * 1000)
+			{
+				long startTime = getElapsedUsecs(tp_rover_start);
+				// Blocks until a newline is found in cin
+				std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+				elapsedTime = getElapsedUsecs(tp_rover_start) - startTime;
+			}
+			autonomous.startNextLeg();
+        }
+
```

Comment:
As discussed above, I think we can avoid all of this code (lines 205-222) using the new `{"type":"autonomous"}` message from the base station. It's better not to block in the main control loop, anyway.